### PR TITLE
Add Amplify deployment to NextJs example [reopened]

### PR DESCRIPTION
**Pull request by guiyom-e**
Reopen PR #363

Starting from `feat/swarmion-with-next` branch, I added an option to deploy the NextJs 13 app to Amplify, with a CDK construct.
See the updated README for more details on setup and use.

Thanks to @alexandrepernin who first deployed a POC of this new deployment option!

image

### CODE REVIEW

File: examples/swarmion-with-next/frontend/next-app/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+- Whenever a branch starts with `feature/`, a preview will be built and deployed.
+  It is possible to update this branch preview configuration in [`branchSettings.ts`](./hosting/settings/branchSettings.ts)
+
+To include the app deployment in an advanced Continuous Deployment pipeline, you may use [webhooks](https://docs.aws.amazon.com/amplify/latest/userguide/webhooks.html). There is no CDK configuration available for now, but some commands are available in the [AWS CLI](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/amplify/create-webhook.html) to automate the process.
+
+## Deploy on Vercel
+
+Another way to deploy your Next.js app is to use the [Vercel Platform](https://vercel.com/new) from the creators of Next.js.
+
+Check out our [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
```
Comment (adriencaccia) [inline examples/swarmion-with-next/frontend/next-app/README.md:69]:
```suggestion
Check out the [Next.js deployment documentation](https://nextjs.org/docs/deployment) for more details.
```

File: examples/swarmion-with-next/frontend/next-app/hosting/bin.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import { AmplifyStack } from './stack';
+
+const app = new cdk.App();
+
+const CLOUDFORMATION_STACK_NAME = 'AmplifyStackSwarmion';
+
+new AmplifyStack(app, CLOUDFORMATION_STACK_NAME, {
+  analyticsReporting: true,
+  description: 'My Swarmion NextJS APP deployed with Amplify',
```
Comment (adriencaccia) [inline examples/swarmion-with-next/frontend/next-app/hosting/bin.ts:11]:
```suggestion
  description: 'Swarmion NextJS APP deployed with Amplify',
```

File: examples/swarmion-with-next/frontend/next-app/hosting/stack.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  branchPreviewConfiguration,
+  MAIN_BRANCH,
+  mainBranchSettings,
+} from './settings/branchSettings';
+import { buildSettings } from './settings/buildSettings';
+import { environmentVariables } from './settings/environmentVariables';
+import { sourceCodeProvider } from './settings/sourceCodeProvider';
+
+const AMPLIFY_APP_NAME = 'NextJS app from CDK';
+const AWS_AMPLIFY_RESOURCE_NAME = 'MyAmplifyAppWithSwarmion';
```
Comment (adriencaccia) [inline examples/swarmion-with-next/frontend/next-app/hosting/stack.ts:21]:
```suggestion
const AWS_AMPLIFY_RESOURCE_NAME = 'AmplifyAppWithSwarmion';
```

File: examples/swarmion-with-next/frontend/next-app/hosting/stack.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    });
+
+    const iManagedPolicy = aws_iam.ManagedPolicy.fromAwsManagedPolicyName(
+      'AdministratorAccess',
+    );
+    role.addManagedPolicy(iManagedPolicy);
+
+    const amplifyApp = new App(this, AWS_AMPLIFY_RESOURCE_NAME, {
+      appName: AMPLIFY_APP_NAME,
+      description: 'My Swarmion NextJS APP deployed with Amplify',
```
Comment (adriencaccia) [inline examples/swarmion-with-next/frontend/next-app/hosting/stack.ts:40]:
```suggestion
      description: 'Swarmion NextJS APP deployed with Amplify',
```

**Review by adriencaccia — CHANGES_REQUESTED** [review]
Great work @guiyom-e 🎉 

Let me know if everything is working as expected, and if it is we can merge 😃

**Comment by guiyom-e** [comment]
> Great work @guiyom-e 🎉
> 
> Let me know if everything is working as expected, and if it is we can merge 😃

Hi @adriencaccia !
I have just tested to deploy swarmion-with-next project with Amplify, it is working properly!

image

image

**Review by adriencaccia — APPROVED** [review]
Great work @guiyom-e 🎉

**Pull request merged**